### PR TITLE
Use data binding & LiveData integration

### LINKED ISSUE #179: onChanged called more than twice
Hello,

I have used LifecycleActivity and Calling AndroidViewModel's MediatorLiveData method but some time it's setValue called multiple times.

### PR DESCRIPTION
**Pull request by yigit**
This CL moves data binding LiveData usage into data binding since it understands it now.

This CL also fixes the problem in fragments where they would get notifications twice for using the adapter lifecycle

Test: existing tests pass w/o a change
Fixes #179

### CODE REVIEW

**Comment by yigit** [comment]
ugh we need to do this circle CI 2.0 migration :/

File: GithubBrowserSample/app/src/main/res/layout/repo_fragment.xml [hunk 1]
```diff
@@ -24,17 +24,17 @@
 
         <import type="com.android.example.github.vo.Resource" />
 
-        <variable
-            name="repo"
-            type="com.android.example.github.vo.Repo" />
+        <import type="androidx.lifecycle.LiveData"/>
```
Comment (JoseAlcerreca) [inline GithubBrowserSample/app/src/main/res/layout/repo_fragment.xml]: This is strange. Normally you'd use LiveData directly without wrapping the type. Can you add a comment explaining why you're doing this?

File: GithubBrowserSample/app/src/main/res/layout/search_fragment.xml [hunk 2]
```diff
@@ -27,22 +27,18 @@
         <import type="com.android.example.github.vo.Status" />
 
         <import type="com.android.example.github.vo.Resource" />
-
-        <variable
-            name="resultCount"
-            type="int" />
-
+        <import type="androidx.lifecycle.LiveData"/>
```
Comment (JoseAlcerreca) [inline GithubBrowserSample/app/src/main/res/layout/search_fragment.xml]: Same, add a comment explaining why you're using LiveData

File: GithubBrowserSample/app/src/main/res/layout/repo_fragment.xml (hunk 1 above)
Comment (yigit) [reply GithubBrowserSample/app/src/main/res/layout/repo_fragment.xml]: not sure what you mean here. it is just and import and then we specify type in the varaible.

**Comment by yigit** [comment]
idk why CLA fails but since I work at Google, i'll ignore it :)

**Pull request merged**